lib/core/src/chain_swap.rs

```diff
@@ -818,18 +828,25 @@ impl ChainSwapHandler {
         )?
         else {
             return Err(PaymentError::Generic {
-                err: format!(
-                    "Unexpected refund tx type returned for incoming Chain swap {}",
-                    swap.id
-                ),
+                err: format!("Unexpected refund tx type returned for incoming Chain swap {id}",),
             });
         };
         let refund_tx_id = bitcoin_chain_service.broadcast(&refund_tx)?.to_string();
 
-        info!(
-            "Successfully broadcast refund for incoming Chain Swap {}, is_cooperative: {is_cooperative}",
-            swap.id
-        );
+        info!("Successfully broadcast refund for incoming Chain Swap {id}, is_cooperative: {is_cooperative}");
+
+        // After refund tx is broadcasted, set the payment state to `RefundPending`. This ensures:
+        // - the swap is not shown in `list-refundables` anymore
```

Comment:
It's my understanding that this PR fixes that the refundable swap is no longer shown in `list_refundables`. I have two questions, after executing a refund:
1. Is the payment shown with the refund_tx_id in `list_payments`?
2. Is it possible to broadcast another tx to effectively bump the fee?

My concern is the swap is no longer refundable if broadcasting the refund with too low a fee

Reply:
>  Is the payment shown with the refund_tx_id in `list_payments`?

Yes.
 
>  Is it possible to broadcast another tx to effectively bump the fee?

Only for cooperative refunds. The non-coop refunds don't have RBF enabled.

However there is no timeout after which cooperative refunds are impossible. The only way I could test non-cooperative refunds was by commenting out the coop refund call. Boltz participated in the coop refund even weeks after a swap failed.

> My concern is the swap is no longer refundable if broadcasting the refund with too low a fee

From what I can tell, this can only happen if
1. Boltz is not reachable for the coop refund and
2. the user chooses a feerate that is too low

IMO this is unlikely because of the 1st point.

Reply:
> Only for cooperative refunds. The non-coop refunds don't have RBF enabled

I tried on main branch to do this, but I get a "No UTXO found" error. Probably because the electrum [script_list_unspent](https://electrumx.readthedocs.io/en/latest/protocol-methods.html#blockchain-scripthash-listunspent) excludes them **_"Any output that is spent in the mempool does not appear"_**

> Boltz participated in the coop refund even weeks after a swap failed

I think Boltz will always try to cooperate with a refund (always if the lockup failed _or_ if the swap expired after they refund themselves). So it's more doomsday handling

Reply:
Tracking in #539